### DISCUSSION TITLE
WebP and KHR_materials_specular

### DISCUSSION THREAD

**Discussion by Mehequanna** [google/filament#7433] [Q&A] [answered]
Hello! Thanks for all the awesome work you all do. I was given a 3D model with [WebP](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_texture_webp/README.md) textures and [KHR_materials_specular](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_materials_specular). When I run gltfviewer it crashes on the WebP textures. As far as I can tell materials specular is ignored.

Are there any plans to add support for these?

**Comment by romainguy** [comment, marked as answer]
KHR_materials_specular is planned yes, WebP not for now (see [extension](https://github.com/KhronosGroup/glTF/blob/main/extensions/2.0/Vendor/EXT_texture_webp/README.md)). WebP shouldn't be very difficult to add but would increase binary size on some platforms.

**Reply by Mehequanna** [reply]
Thanks @romainguy !

**Reply by Mehequanna** [reply]
I had a follow up question from my 3D lead. Would the binary size increase by kilobytes or megabytes?

**Reply by romainguy** [reply]
On Android there would be no increase potentially. On iOS it looks like it could be megabytes.

**Reply by Mehequanna** [reply]
Thank you again!

**Reply by hannojg** [reply]
See this for webp: https://github.com/google/filament/issues/7962

**Reply by PsycoTodd** [reply]
KHR_material_specular is also added.
